app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -63,68 +64,83 @@ class SuggestedEditsContributionsFragment : Fragment(), SuggestedEditsTypeItem.C
+                } else if (scrollY != 0 && activity.supportActionBar?.elevation == 0f) {
+                    activity.supportActionBar?.elevation = DimenUtil.dpToPx(DimenUtil.getDimension(R.dimen.toolbar_default_elevation))
                 }
             }
+        })
+
+        swipeRefreshLayout.setColorSchemeResources(ResourceUtil.getThemedAttributeId(requireContext(), R.attr.colorAccent))
+        swipeRefreshLayout.setOnRefreshListener {
+            resetAndFetch()
         }
-        contributionsRecyclerView.addOnScrollListener(scrollListener)
+
+        resetAndFetch()
+    }
+
+    override fun onDestroyView() {
+        contributionsRecyclerView.adapter = null
+        contributionsRecyclerView.clearOnScrollListeners()
+        disposables.clear()
+        super.onDestroyView()
     }
 
-    private fun setFilterAndUIState(view: SuggestedEditsTypeItem) {
-        editFilterType = view.editType
+    override fun onTypeItemClick(editType: Int) {
+        editFilterType = editType
         createConsolidatedList()
     }
 
     companion object {
+        private const val VIEW_TYPE_HEADER = 0
+        private const val VIEW_TYPE_DATE = 1
+        private const val VIEW_TYPE_ITEM = 2
+
         fun newInstance(): SuggestedEditsContributionsFragment {
             return SuggestedEditsContributionsFragment()
         }
     }
 
-    private fun loadContentOfEditFilterType() {
-        loadingMore = true
-        if (editFilterType == ALL_EDIT_TYPES || editFilterType == EDIT_TYPE_ARTICLE_DESCRIPTION) {
-            getArticleContributions()
-        } else {
-            getImageContributions()
-        }
+    private fun resetAndFetch() {
+        allContributions.clear()
+        displayedContributions.clear()
+        articleContributionsContinuation = null
+        imageContributionsContinuation = null
+        adapter.notifyDataSetChanged()
+        fetchContributions()
     }
 
-    private fun getArticleContributions() {
-        val qLangMap = HashMap<String, HashSet<String>>()
-        disposables.add(ServiceFactory.get(WikiSite(Service.WIKIDATA_URL)).getUserContributions(AccountUtil.getUserName()!!, 50, articleContributionsContinuation)
+    private fun fetchContributions() {
+        if (allContributions.isNotEmpty() && articleContributionsContinuation.isNullOrEmpty() && imageContributionsContinuation.isNullOrEmpty()) {
+            // there's nothing more to fetch!
+            return
+        }
+
+        progressBar.visibility = VISIBLE
+        disposables.clear()
+
+        if (allContributions.isEmpty()) {
+            disposables.add(SuggestedEditsUserStats.getPageViewsObservable().subscribe {
+                totalPageViews = it
+                adapter.notifyDataSetChanged()
+            })
+        }
+
+        disposables.add(Observable.zip(if (allContributions.isNotEmpty() && articleContributionsContinuation.isNullOrEmpty()) Observable.just(Collections.emptyList<Contribution>())
```

Comment:
Explicit type argument can be removed.

Reply:
I don't think it can. Try it for yourself and see if the build succeeds.

Reply:
You are right. How about just use Observable.empty()?